outputs.tf

```
@@ -14,8 +14,8 @@ output "identity_pool_id" {
 }
 
 output "client_secret_secret_id" {
-  value       = aws_secretsmanager_secret.client_secret.id
-  description = "string ||| The secret id for the AWS secrets manager secret that contains the client_secret for the cognito user pool."
+  value       = var.client_secret != "" ? aws_secretsmanager_secret.client_secret.id : ""
```

Add explicit indexing
```suggestion
  value       = var.client_secret != "" ? aws_secretsmanager_secret.client_secret[0].id : ""
```